drivers/power/lt8491/lt8491.c

```diff
@@ -0,0 +1,612 @@
+ * @param dev - LT8491 descriptor
+ * @param val - telemetry battery efficiency in m%
+ * @return 0 in case of success, errno errors otherwise
+ */
+int lt8491_read_efficiency(struct lt8491_device *dev, uint32_t *val)
+{
+	int ret;
+	uint16_t uval;
+
+	ret = __lt8491_update_telemetry(dev);
+	if (ret)
+		return ret;
+
+	ret = lt8491_reg_read(dev, LT8491_TELE_EFF_REG, &uval, true);
+	if (ret)
+		return ret;
+
+	*val = uval * 10;
+
+	return 0;
+}
+
+/**
+ * @brief Read the telemetry battery output current
+ * @param dev - LT8491 descriptor
+ * @param val - telemetry battery output current in mA
+ * @return 0 in case of success, errno errors otherwise
+ */
+int lt8491_read_iout(struct lt8491_device *dev, uint32_t *val)
+{
+	int ret;
+	uint16_t uval;
+
+	ret = __lt8491_update_telemetry(dev);
+	if (ret)
+		return ret;
+
+	ret = lt8491_reg_read(dev, LT8491_TELE_IOUT_REG, &uval, true);
+	if (ret)
+		return ret;
+
+	*val = uval;
+
+	return 0;
+}
+
+/**
+ * @brief Read the telemetry battery input current
+ * @param dev - LT8491 descriptor
+ * @param val - telemetry battery input current in mA
+ * @return 0 in case of success, errno errors otherwise
+ */
+int lt8491_read_iin(struct lt8491_device *dev, uint32_t *val)
+{
+	int ret;
+	uint16_t uval;
+
+	ret = __lt8491_update_telemetry(dev);
+	if (ret)
+		return ret;
+
+	ret = lt8491_reg_read(dev, LT8491_TELE_IIN_REG, &uval, true);
+	if (ret)
+		return ret;
+
+	*val = uval;
+
+	return 0;
+}
+
+/**
+ * @brief Read the telemetry battery voltage
+ * @param dev - LT8491 descriptor
+ * @param val - telemetry battery voltage in mV
+ * @return 0 in case of success, errno errors otherwise
+ */
+int lt8491_read_vbat(struct lt8491_device *dev, uint32_t *val)
+{
+	int ret;
+	uint16_t uval;
+
+	ret = __lt8491_update_telemetry(dev);
+	if (ret)
+		return ret;
+
+	ret = lt8491_reg_read(dev, LT8491_TELE_VBAT_REG, &uval, true);
+	if (ret)
+		return ret;
+
+	*val = uval * 10;
+
+	return 0;
+}
+
+/**
+ * @brief Read the telemetry input voltage
+ * @param dev - LT8491 descriptor
+ * @param val - telemetry input voltage in mV
+ * @return 0 in case of success, errno errors otherwise
+ */
+int lt8491_read_vin(struct lt8491_device *dev, uint32_t *val)
+{
+	int ret;
+	uint16_t uval;
+
+	ret = __lt8491_update_telemetry(dev);
+	if (ret)
+		return ret;
+
+	ret = lt8491_reg_read(dev, LT8491_STAT_SUPPLY_REG, &uval, false);
+	if (ret)
+		return ret;
+
+	if (no_os_field_get(LT8491_PS_OR_SOLAR_MASK, uval))
+		ret = lt8491_reg_read(dev, LT8491_TELE_VIN_REG, &uval, true);
+	else
+		ret = lt8491_reg_read(dev, LT8491_TELE_VINR_REG, &uval, true);
+	if (ret)
+		return ret;
+
+	*val = uval * 10;
+
+	return 0;
+}
+
+/**
+ * @brief Read the telemetry battery charging stage
+ * @param dev - LT8491 descriptor
+ * @param charval - char to store telemetry battery charging stage
+ * @return 0 in case of success, errno errors otherwise
+ */
+int lt8491_read_charging_stage(struct lt8491_device *dev, int32_t *charval)
```

Comment:
You could just use the enum values in an application directly. Why is there a need to assign a character to them?